scout/server/blueprints/institutes/views.py

```diff
@@ -89,3 +90,64 @@ def institute(institute_id):
     return render_template(
         "/overview/institute.html", form=form, default_phenotypes=default_phenotypes, **data
     )
+
+
+@blueprint.route("/<institute_id>/clinvar_submissions", methods=["GET", "POST"])
+@templated("overview/clinvar_submissions.html")
+def clinvar_submissions(institute_id):
+    """Handle clinVar submission objects and files"""
+
+    def generate_csv(header, lines):
+        yield header + "\n"
+        for line in lines:  # lines have already quoted fields
+            yield line + "\n"
+
+    if request.method == "POST":
+        submission_id = request.form.get("submission_id")
+        if request.form.get("update_submission"):
+            controllers.update_clinvar_submission_status(
+                store, request, institute_id, submission_id
+            )
+        elif request.form.get("delete_variant"):  # delete a variant from a submission
+            store.delete_clinvar_object(
+                object_id=request.form.get("delete_variant"),
+                object_type="variant_data",
+                submission_id=submission_id,
+            )  # remove variant and associated_casedata
+        elif request.form.get("delete_casedata"):  # delete a case from a submission
+            store.delete_clinvar_object(
+                object_id=request.form.get("delete_casedata"),
+                object_type="case_data",
+                submission_id=submission_id,
+            )  # remove just the casedata associated to a variant
+        elif request.form.get("newSampleName"):
+            # Rename one or more casedata samples
+            controllers.update_clinvar_sample_names(
+                store,
+                submission_id,
+                request.form.get("oldSampleName"),
+                request.form["newSampleName"],
+            )
+        else:
```

Comment:
This does sort of have the consequence (intended or not) that clicking or hitting enter a bit randomly on the page triggers download. But, worse things could happen! 😺

Reply:
Solved by refactoring!